### PR TITLE
aws - elasticache and rg skip delete when linked with global ds

### LINKED ISSUE #8815: aws - cache-cluster - error occurred (InvalidParameterValue)
### Describe the bug

When trying to delete a cache-cluster resource type, the operation fails when the cache-cluster's replication group is part of a global cluster. Traceback

### What did you expect to happen?

Log a warning and proceed?

### Cloud Provider

Amazon Web Services (AWS)

### Cloud Custodian version and dependency information

```shell
Custodian:   0.9.28
Python:      3.11.4 (main, Jul 25 2023, 17:36:13) [Clang 14.0.3 (clang-1403.0.22.14.1)]
```

### Policy

```shell
- name: cache-cluster-minor-version-existing-delete
  resource: aws.cache-cluster
  filters:
  - type: value
    key: AutoMinorVersionUpgrade
    op: not-equal
    value: true
  actions:
  - type: delete
```

### Relevant log/traceback output

```shell
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/c7n/policy.py", line 364, in run
    results = a.process(resources)
  File "/usr/local/lib/python3.9/dist-packages/c7n/resources/elasticache.py", line 170, in process
    client.delete_replication_group(**params)
  File "/usr/local/lib/python3.9/dist-packages/botocore/client.py", line 530, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/local/lib/python3.9/dist-packages/botocore/client.py", line 964, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.InvalidParameterValueException: An error occurred (InvalidParameterValue) when calling the DeleteReplicationGroup operation: Cluster [xyz] is part of a global cluster [xyz]. Request rejected.
```

### Extra information or context

_No response_

### PR DESCRIPTION
**Pull request by PratMis**
Closes #8815

### CODE REVIEW

File: c7n/resources/elasticache.py [hunk 1]
```diff
@@ -514,9 +524,23 @@ class DeleteReplicationGroup(BaseAction):
...
+                self.log.info(
+                  f"Skipping {r['ReplicationGroupId']}: associated with a global datastore")
+                continue
             params = {'ReplicationGroupId': r['ReplicationGroupId']}
             if self.data.get('snapshot', False):
                 params.update({'FinalSnapshotIdentifier': r['ReplicationGroupId'] + '-snapshot'})
             self.manager.retry(client.delete_replication_group, **params, ignore_err_codes=(
                 'ReplicationGroupNotFoundFault',))
+
+    def get_gloabal_datastore_association(self, resources):
```
Comment (kapilt) [inline c7n/resources/elasticache.py:539]: typo in global

Comment (PratMis) [reply c7n/resources/elasticache.py]: oops, sorry. It's fixed now

**Comment by kapilt** [comment]
you might want a base class mixing for the shared bits, re test failure.

**Comment by PratMis** [comment]
> you might want a base class mixing for the shared bits, re test failure.

Looks like I still had more more typo using the old function name. Changing that fixed the test failure.

File: c7n/resources/elasticache.py [hunk 2]
```diff
@@ -173,6 +177,12 @@ def process(self, clusters):
                 'Deleted ElastiCache replication group: %s',
                 replication_group)
 
+    def fetch_global_ds_rpgs(self):
+        global_rpgs = set()
```
Comment (kapilt) [inline c7n/resources/elasticache.py:181]: this line is superfluous afaics

File: c7n/resources/elasticache.py [hunk 3]
```diff
@@ -173,6 +177,12 @@ def process(self, clusters):
...
+        rpgs = self.manager.get_resource_manager('elasticache-group').resources()
```
Comment (kapilt) [inline c7n/resources/elasticache.py:182]: augment=False we don't need tags here.

Comment (PratMis) [reply c7n/resources/elasticache.py]: Thanks, good catch!! Both the feedbacks should be addressed now.

File: c7n/resources/elasticache.py (hunk 2 above)
Comment (PratMis) [reply c7n/resources/elasticache.py]: Thanks for the feedback @kapilt

**Review by kapilt — APPROVED** [review]
lgtm

**Pull request merged**